### PR TITLE
Move DefaultWebProxySettings from Common to each package that was using it

### PR DESCRIPTION
**Pull request by timtay-microsoft**
~~This class served as a stubbed out IWebProxy implementation that we used as the value to signal that no proxy is configured. To simplify this, we can use "null" as that value instead and delete this class~~

To preserve the currently functionality outlined by @abhipsaMisra below, I've opted to simply clone this DefaultProxySettings class to each library that was using it.

### CODE REVIEW

**Comment by abhipsaMisra** [comment]
Make sure that all three scenarios are still supported:

> Currently, the sdk picks up the proxy settings as per the OS default proxy settings. With this change, we will be exposing an IWebProxy member to all our IoT SDK clients, allowing for setting the proxy explicitly in the application.
>
> The following 3 scenarios will be supported:
>
> Proxy not set in the code (sdk will pick the OS default proxy settings)
> Proxy = null (sdk will not use proxy)
> Proxy = set in the code (sdk will use custom proxy specified).

https://github.com/Azure/azure-iot-sdk-csharp/pull/569

**Comment by timtay-microsoft** [comment]
> Make sure that all three scenarios are still supported:
> 
> > Currently, the sdk picks up the proxy settings as per the OS default proxy settings. With this change, we will be exposing an IWebProxy member to all our IoT SDK clients, allowing for setting the proxy explicitly in the application.
> > The following 3 scenarios will be supported:
> > Proxy not set in the code (sdk will pick the OS default proxy settings)
> > Proxy = null (sdk will not use proxy)
> > Proxy = set in the code (sdk will use custom proxy specified).
> 
> #569

Interesting... I didn't know that we supported a "Let the OS decide" proxy setting here. Let me rework this a bit then

File: iothub/service/src/Http2/HttpClientFactory.cs [hunk 1]
```diff
@@ -53,9 +53,9 @@ internal static HttpClient Create(string hostName, IotHubServiceClientOptions op
             httpMessageHandler.SslProtocols = options.SslProtocols;
             httpMessageHandler.CheckCertificateRevocationList = options.CertificateRevocationCheck;
 
-            if (options.Proxy != DefaultWebProxySettings.Instance)
+            if (options.Proxy != null)
```
Comment (timtay-microsoft) [inline iothub/service/src/Http2/HttpClientFactory.cs:56]:
```suggestion
            if (options.Proxy != DefaultWebProxySettings.Instance)
```

File: iothub/service/src/Http2/HttpClientFactory.cs [hunk 2]
```diff
@@ -55,7 +55,7 @@ internal static HttpClient Create(string hostName, IotHubServiceClientOptions op
 
             if (options.Proxy != DefaultWebProxySettings.Instance)
             {
-                httpMessageHandler.UseProxy = options.Proxy != null;
+                httpMessageHandler.UseProxy = true;
```
Comment (timtay-microsoft) [inline iothub/service/src/Http2/HttpClientFactory.cs:58]:
```suggestion
                httpMessageHandler.UseProxy = options.Proxy != null;
```

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**